### ISSUE TITLE
Deploy an individual step-function

### ISSUE THREAD

**Issue by ksnyde** [horike37/serverless-step-functions#112]
There appears to be no way to deploy a single step-function (or set of step-functions). This forces a full deployment which in a large project is brutally slow. I guess I don't precisely what the limitations are here but it would **SUPER** nice to get this capability into the deployment functions.

**Comment by lucas-rudd** [comment]
@ksnyde this is a limitation of Serverless and CloudFormation I believe. Since all of this gets transpiled into CloudFormation as a stack, and all the Resources in that stack are managed _by_ that stack. You could create nested directory structures to handle this functionality.

For example
```
mainProject/
├── stepFunction1/
│   ├── src/
│  │        ├── index.js
│   ├── serverless.yml
 ├── stepFunction2/
│   ├── src/
│  │        ├── index.js
│   ├── serverless.yml
```
etc.

The Stacks created by `serverless deploy` will be different when created in each directory, and you can deploy each stack independently.

**Comment by ksnyde** [comment]
@lucas-rudd sorry just seeing this now ... question ... if I do this do i lose the ability to map/link the step functions to Tasks (aka, lambda functions) in the main project?

**Comment by hotgazpacho** [comment]
Serverless plugins get access to the AWS SDK; might it be possible to add a command that compiles the ASL JSON and uses [updateStateMachine](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/StepFunctions.html#updateStateMachine-property) to update the definition of the state machine independently of CloudFormation? 

Of course, put huge caveats on it that this command is intended as a development-time optimization only.

Thoughts?

**Comment by theburningmonk** [comment]
@hotgazpacho it's definitely doable, although I'm not sure if it's worth all the effort. If it's purely to optimize for development time and have fast feedback loop, I think you're better off just doing that in the AWS console, that way you get the visualization tool as well.

**Comment by hotgazpacho** [comment]
@theburningmonk I agree that the console is great for visualizing and initial prototyping. However, the problem comes when I need to make changes to a workflow, maybe add some steps. I can’t simply copy my workflow definition into the console and have it run; the ARNs for the resources in each task are interpolated by this plugin at deploy time. Serverless already lets me deploy lambda functions independently; it would be hugely helpful for development to do the same with the step function definitions.

**Comment by theburningmonk** [comment]
@hotgazpacho If you're happy to put together a PR, then of course it'll be considered. But bear in mind that there are quite a few edge cases you have to cover to make this feature actually useful. If you just do `updateStateMachine` then lots of things won't work and it'll be a surprise to anyone using the feature. You'll need to update the relevant IAM roles, API Gateway configurations, etc. 

For example,  if the user adds a new `Task` step (against a new function that user might have deployed with `sls deploy -f`), and the user hasn't specified an IAM role override, then you'll need to also update the IAM role as well. Otherwise, the updated state machine would not have the necessary permissions to invoke the new function.

If the feature can only work in narrow cases then I'm against putting it in. But if you find a way to make it work fairly consistently (happy to ignore things like alerts and notifications) then I'm all for it.

**Comment by hotgazpacho** [comment]
Oh wow, that _is_ a lot of cases to consider! That’s probably more than I can commit to tackling at this time. Thank you for elaborating on this; great insight!
